R/methods-epi_archive.R

```diff
@@ -1000,3 +1000,79 @@ epix_truncate_versions_after.epi_archive <- function(x, max_version) {
   return((x$clone()$truncate_versions_after(max_version)))
   # ^ second set of parens drops invisibility
 }
+
+#' Print and summary functions for an `epi_archive` object.
+#'
+#' @param x The `epi_archive` object.
+#' @param class Boolean; whether to print the class label header
+#' @param methods Boolean; whether to print all available methods of
+#'   the archive
+#' @param ... Additional arguments passed to methods.
+#'
+#' @method print epi_archive
+#' @export
+print.epi_archive <- function(x, class = TRUE, methods = TRUE, ...) {
+  # --- Copied from R6 ---
+  if (class) cat("An `epi_archive` object, with metadata:\n")
+  cat(sprintf("* %-9s = %s\n", "geo_type", x$geo_type))
+  cat(sprintf("* %-9s = %s\n", "time_type", x$time_type))
+  if (!is.null(x$additional_metadata)) {
+    sapply(x$additional_metadata, function(m) {
+      cat(sprintf("* %-9s = %s\n", names(m), m))
+    })
+  }
+  cat("----------\n")
+  if (length(x$DT$time_value) == 0 || all(is.na(x$DT$time_value))) {
+    min_time <- max_time <- NA
+  } else {
+    min_time <- Min(x$DT$time_value)
+    max_time <- Max(x$DT$time_value)
+  }
+  cat(sprintf("* %-14s = %s\n", "min time value", min_time))
+  cat(sprintf("* %-14s = %s\n", "max time value", max_time))
+  cat(sprintf(
+    "* %-14s = %s\n", "first version with update",
+    min(x$DT$version)
+  ))
+  cat(sprintf(
+    "* %-14s = %s\n", "last version with update",
+    max(x$DT$version)
+  ))
+  if (is.na(x$clobberable_versions_start)) {
+    cat("* No clobberable versions\n")
+  } else {
+    cat(sprintf(
+      "* %-14s = %s\n", "clobberable versions start",
+      x$clobberable_versions_start
+    ))
+  }
+  cat(sprintf(
+    "* %-14s = %s\n", "versions end",
+    x$versions_end
+  ))
+  cat("----------\n")
+  cat(sprintf(
+    "Data archive (stored in DT field): %i x %i\n",
+    nrow(x$DT), ncol(x$DT)
+  ))
+  cat(sprintf("Columns in DT: %s\n", paste(ifelse(length(
+    colnames(x$DT)
+  ) <= 4, paste(colnames(x$DT), collapse = ", "),
+  paste(
+    paste(colnames(x$DT)[1:4], collapse = ", "), "and",
+    length(colnames(x$DT)[5:length(colnames(x$DT))]), "more columns"
+  )
+  ))))
+  if (methods) {
+    cat("----------\n")
+    writeLines(wrap_varnames(
+      initial = "Public R6 methods: ",
+      names(epi_archive$public_methods)
+    ))
+  }
```

Comment:
This can probably go, as we're moving away from R6, yea?

Reply:
It should go away when we're actually doing the move from R6.  And maybe be replaced by some alternative guidance as to what you can actually do with archives.  Think this is going to be easier to do all at once later on.

Reply:
Sounds good